mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -1526,12 +1532,15 @@ static LogicalResult populateValidationLogic(
       builder.create<DeallocOp>(builder.getUnknownLoc(), inputHostAllocOp);
   auto outputHostDeallocOp =
       builder.create<DeallocOp>(builder.getUnknownLoc(), outputHostAllocOp);
-  auto cpuResultsDeallocOp =
-      builder.create<DeallocOp>(builder.getUnknownLoc(), cpuResults);
   block->push_back(filterHostDeallocOp);
   block->push_back(inputHostDeallocOp);
   block->push_back(outputHostDeallocOp);
-  block->push_back(cpuResultsDeallocOp);
+
+  if (dataType != builder.getIntegerType(16)) { // i16 dealloc in below
+    auto cpuResultsDeallocOp =
+        builder.create<DeallocOp>(builder.getUnknownLoc(), cpuResults);
+    block->push_back(cpuResultsDeallocOp);
+  }
```

Comment:
use `if / else` instead of 2 `if` blocks.

Reply:
fixed